tests/test-core/utils/js/ConditionalTestUtils.js

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+
+*/
+
+/* global fluid, jqUnit */
+
+(function () {
+    "use strict";
+
+    fluid.registerNamespace("fluid.tests.conditionalTestUtils");
+
+    // "platformArg" is a string or array of strings of platform names to detect
+    // This returns a boolean based on whether or not the string is
+    // present in the navigator.platform string
+
+    fluid.tests.conditionalTestUtils.isPlatform = function (platformArg) {
+        var detectedPlatform = navigator.platform;
```

Comment:
This will also aid you in writing test cases for this facility itself - otherwise it will be rather hard to mock!